lib/events/filesessions/fileasync.go

```diff
@@ -128,49 +129,125 @@ type Uploader struct {
 	log             *log.Entry
 	uploadCompleter *events.UploadCompleter
 
-	cancel context.CancelFunc
-	ctx    context.Context
+	cancel   context.CancelFunc
+	ctx      context.Context
+	eventsCh chan events.UploadEvent
+}
+
+func (u *Uploader) writeSessionError(sessionID session.ID, err error) error {
+	if sessionID == "" {
+		return trace.BadParameter("missing session ID")
+	}
+	path := u.sessionErrorFilePath(sessionID)
+	return trace.ConvertSystemError(ioutil.WriteFile(path, []byte(err.Error()), 0600))
+}
+
+func (u *Uploader) checkSessionError(sessionID session.ID) (bool, error) {
+	if sessionID == "" {
+		return false, trace.BadParameter("missing session ID")
+	}
+	_, err := os.Stat(u.sessionErrorFilePath(sessionID))
+	if err != nil {
+		err = trace.ConvertSystemError(err)
+		if trace.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
 }
 
 // Serve runs the uploader until stopped
```

Comment:
Explain how the Uploader works (including corrupt session handling) in the comment here or on the Uploader struct.